subiquity/cloudinit.py

```diff
@@ -0,0 +1,12 @@
+"""Shared cloudinit utility functions"""
+
+import json
+
+
+def get_host_combined_cloud_config() -> dict:
+    """Return the host system /run/cloud-init/combined-cloud-config.json"""
+    try:
+        with open("/run/cloud-init/combined-cloud-config.json") as fp:
+            return json.load(fp)
```

Comment:
Do we want to log success here so we can track any call-sites since models/network.py also re-reads this file. If we grow lots of readers, we may look toward `lru_cache` of this function in the future even though it's not very expensive.
```suggestion
            config = json.load(fp)
        log.debug("loading cloud config from /run/cloud-init/combined-cloud-config.json")
```

Reply:
Done